frontend/public/components/machine-set.tsx

```diff
@@ -299,11 +363,39 @@ export const MachineSetList: React.SFC = (props) => {
               obj.metadata.namespace,
             )}/machines`}
           >
-            {getReadyReplicas(obj)} {t('public~of')} {getDesiredReplicas(obj)}{' '}
-            {t('public~machines')}
+            {t('public~{{numReplicas}} of {{numDesiredReplicas}} machine', {
+              numReplicas: getReadyReplicas(obj),
+              numDesiredReplicas: getDesiredReplicas(obj),
+              count: getDesiredReplicas(obj),
+            })}
           </Link>
         </TableData>
         <TableData className={tableColumnClasses[3]}>
+          {getMachineSetInstanceType(obj) || '-'}
+        </TableData>
+        <TableData className={tableColumnClasses[4]}>
+          {relatedNodes.reduce(
+            (cpuSize, node) => cpuSize + convertToBaseValue(node.status?.capacity.cpu),
```

Comment:
You use optional chaining, but if `node.status` is undefined, `cpuSize` will be converted to `NaN`. If `node.status` is not always set, we need to handle it differently.

Reply:
Thanks! This is a good point. I'll switch over to using _.toInteger. I tried it and it will convert NaN, undefined, null, etc. to 0 so we'd get the expected behavior.